src/flag_gems/ops/select_scatter.py

```diff
@@ -1,13 +1,45 @@
 import logging
 
 import torch
+import triton
+import triton.language as tl
 
-from flag_gems.ops.copy import copy
 from flag_gems.utils.shape_utils import MemOverlap, has_internal_overlapping
 
 logger = logging.getLogger(__name__)
 
 
+@triton.jit
+def select_scatter_kernel(
+    out_ptr,
+    inp_ptr,
+    src_ptr,
+    total_elements,
+    dim_size,
+    dim_prod_post,
+    index,
+    BLOCK_SIZE: tl.constexpr,
+):
+    pid = tl.program_id(0)
+    block_start = pid * BLOCK_SIZE
+    offsets = tl.arange(0, BLOCK_SIZE)
+    mask = block_start + offsets < total_elements
+    idx = block_start + offsets
+
+    pre_idx = idx // (dim_size * dim_prod_post)
+    dim_idx = (idx // dim_prod_post) % dim_size
+    post_idx = idx % dim_prod_post
+
+    select_mask = dim_idx == index
+
+    inp_data = tl.load(inp_ptr + idx, mask=mask)
```

Comment:
Can the mask here be avoided? Since loading it without masking may have better performance.